File: python-examples/hyprid-automation/utils/crawler/content.py
from intuned_browser import extract_markdown
from intuned_browser.ai import extract_structured_data
from playwright.async_api import Page


async def extract_page_content(
    page: Page,
    schema: dict | None = None,
) -> dict:
    if schema:
        return await extract_structured_data(source=page, data_schema=schema, model="gpt-5-mini")

    title = await page.title()
    markdown = await extract_markdown(page)

    return {
        "title": title,
        "markdown": markdown,
        "markdown_length": len(markdown) if markdown else 0,
    }

File: typescript-examples/hybrid-automation/utils/crawler/content.ts
import { Page } from "playwright";
import { extractMarkdown } from "@intuned/browser";
import { extractStructuredData } from "@intuned/browser/ai";

export async function extractPageContent(
  page: Page,
  schema?: any
): Promise<any> {
  if (schema) {
    return await extractStructuredData({
      source: page,
      dataSchema: schema,
      model: "gpt-5-mini",
    });
  }

  const title = await page.title();
  const markdown = await extractMarkdown({ source: page });

  return {
    title,
    markdown,
    markdown_length: markdown?.length || 0,
  };
}
